src/v/kafka/server/handlers/incremental_alter_configs.cc

```diff
@@ -54,7 +54,36 @@ void parse_and_set_optional(
     }
     // set property value
     if (op == config_resource_operation::set) {
-        property.value = boost::lexical_cast<T>(*value);
+        property.value = boost::lexical_cast<t>(*value);
+        property.op = cluster::incremental_update_operation::set;
+        return;
+    }
+}
+
+void parse_and_set_bool(
+  cluster::property_update<bool>& property,
+  const std::optional<ss::sstring>& value,
+  config_resource_operation op,
+  bool default_value) {
+    // A remove on a concrete (non-nullable) property is a reset to default,
+    // as is an assignment to nullopt.
+    if (
+      op == config_resource_operation::remove
+      || (op == config_resource_operation::set && !value)) {
+        property.op = cluster::incremental_update_operation::set;
+        property.value = default_value;
+        return;
+    }
+
+    if (op == config_resource_operation::set && value) {
+        try {
+            property.value = string_switch<bool>(*(value))
+                               .match("true", true)
+                               .match("false", false);
+        } catch (...) {
```

Comment:
Yes, although it's still quite ugly (string_switch throws runtime_error rather than something nicely typed).  Still, you're right, it's not like I meant to catch e.g. bad_alloc here.  Fixed.